### ISSUE TITLE
too many matching process interruptions

### ISSUE THREAD

**Issue by blaueente** [mh-/corona-warn-companion-android#152]
Now at 3 million diagnostic keys the matching takes very long. 
This is of course not optimal, but I can live with my device calculating for a while, I understand there are reasons why that is ( https://github.com/mh-/corona-warn-companion-android/issues/149 )
The real annoyance is having to "babysit" it during that increasing timespan:

1. Start CWC
2. wait
3. touch to reset screenlock timer
4. wait 
5. touch
6. ...
100. results

Problem 1: this requires continous active touching, otherwise often the screenlock turns on quite early, and the progress will be lost and cwc locks up. 
Problem 2: Alternative: screen split would enable to e.g. use other apps while calculating, but this does not work. cwc locks up
Problem 3: Alternative put cwc into background, similar, but cwc locks up

When cwc locks up, it gets non-responsive, displays blank or the logo, and I need to go to task list and swipe it up.

Most optimal would be of course keep matching in background and just giving a notification upon finishing, but even very simple options would improve this situation a lot:  e.g. keep screen on until calculation is finished? or pause and resume calculation?

**Comment by mh-** [comment]
a) which app version are you using?

b) are you aware that you can reduce the number of download days?

**Comment by blaueente** [comment]
a) 2.3.2 from the playstore
b) yes, I am aware. This does shorten the time, but the overall problem is not really solved.

**Comment by mh-** [comment]
This is really strange, because 2.3.2 includes https://github.com/mh-/corona-warn-companion-android/commit/41c0191937e229903665552cb1a48c22695d306c (`android:keepScreenOn="true"` for the main activity), which should keep the screen active. During my tests, the screen saver did _not_ kick in.

**Comment by blaueente** [comment]
Indeed, as I tried it again just now, it doesn't kick in. ProbablyI mixed up the versions, as I played around with the git master to understand the behaviour before submitting the bug, and didn't re-test all of my problems with the new install.
My bad :(
The other problems with the crashes are still present though, but now at least I can leave it calculating unattended.

**Comment by skyerjoe** [comment]
>  The other problems with the crashes are still present though, but now at least I can leave it calculating unattended.

I could confirm the crashes with a high day intervall. The matching with 3 Millions Keys cost a bunch of Power so as @mh already saifd try to decrese the intervall. 

best regards
